### PR TITLE
Docs usage with next.js, add information about workaround for Next.js 14 and App Router

### PR DESCRIPTION
**Pull request by pavelee**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/45567

### 💡 Background and solution

Updating official docs about workaround with Next 14 to avoid UI flickering 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     --      |
| 🇨🇳 Chinese |     --      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at ed12e10</samp>

Improved the documentation for using Ant Design with Next.js in both English and Chinese. Fixed some issues with style loading and compatibility with Next.js 14.

### <samp>🤖 Generated by Copilot at ed12e10</samp>

* Move comment about `use client` to a separate line for better readability and consistency ([link](https://github.com/ant-design/ant-design/pull/45642/files?diff=unified&w=0#diff-f40a07db5e50bc714a69751497cc62a602271f477eabf7c895f2fa9950461ae4L36-R38), [link](https://github.com/ant-design/ant-design/pull/45642/files?diff=unified&w=0#diff-b532268d5c5f44bba9caadf328fe6567e31ba5fc7c36cfe6eec648bcf1b36489L36-R38))
* Add comment to inform users of Next.js 14 to use a different import for `StyleProvider` due to compatibility issue and provide link to GitHub issue ([link](https://github.com/ant-design/ant-design/pull/45642/files?diff=unified&w=0#diff-f40a07db5e50bc714a69751497cc62a602271f477eabf7c895f2fa9950461ae4R70-R71), [link](https://github.com/ant-design/ant-design/pull/45642/files?diff=unified&w=0#diff-b532268d5c5f44bba9caadf328fe6567e31ba5fc7c36cfe6eec648bcf1b36489R70-R71))

### CODE REVIEW

File: docs/react/use-with-next.en-US.md [hunk 1]
```diff
@@ -33,8 +33,9 @@ Now we install `antd` from yarn or npm or pnpm.
 Modify `src/app/page.tsx`, import Button component from `antd`.
 
 ```tsx
-'use client'; // If used in Pages Router, is no need to add "use client"
+'use client';
```
Comment (li-jia-nan) [inline docs/react/use-with-next.en-US.md:36]: The format here should not change

Comment (li-jia-nan) [reply docs/react/use-with-next.en-US.md:36]: So that's it, you can manually edit on GitHub to revert this line of code

Comment (pavelee) [reply docs/react/use-with-next.en-US.md]: Done, thanks for the tip 🙏

Comment (li-jia-nan) [reply docs/react/use-with-next.en-US.md]: You're welcome, and I need to verify if the same problem exists on the Pages Router.

Comment (pavelee) [reply docs/react/use-with-next.en-US.md]: I did that and I didn't notice the issue on the Pages Router

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**